homeassistant/components/homekit/__init__.py

```diff
@@ -120,11 +116,11 @@ def get_accessory(hass, state, aid, config):
             or state.domain == 'input_boolean' or state.domain == 'script':
         a_type = 'Switch'
 
-    if a_type is None:
+    else:
```

Comment:
This change will cause issues....
state.domain == 'sensor' and unit is not %, or temp and a_type is undefined

Please revert to the original `if a_type is None:`

Reply:
You're right. Didn't thought about that.